Sources/NIO/CircularBuffer.swift

```diff
@@ -79,6 +85,87 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
         }
     }
 
+    /// Replaces the specified subrange of elements with the given collection.
+    ///
+    /// - Parameter subrange:
+    /// The subrange of the collection to replace. The bounds of the range must be valid indices of the collection.
+    ///
+    /// - Parameter newElements:
+    /// The new elements to add to the collection.
+    ///
+    /// *O(n)* where _n_ is the lenght of the new elements collection if the subrange equals to _n_
+    ///
+    /// *O(m)* where _m_ is the combined length of the collection and _newElements_
+    public mutating func replaceSubrange(_ subrange: RangeType<Int>, with newElements: ContiguousArray<E>) {
```

Comment:
Does any of this function operate on `ContiguousArray` directly? It doesn't seem like it does, in which case it seems like it'd be easier to just make this function generic over `Collection<E>`.